src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -121,6 +145,36 @@ void GcsServer::Start() {
   gcs_init_data->AsyncLoad([this, gcs_init_data] { DoStart(*gcs_init_data); });
 }
 
+void GcsServer::CacheAndSetServerToken() {
+  static std::string const kTokenNamespace = "cluster";
```

Comment:
huh, do we need to differentiate the in_memory vs persist storage? I thought we can always assume that the cluster is new if the token is missing.

Reply:
Yes, one codepath covers both, but checking and directly setting is an optimization. Up to you which one makes more sense.